src/vivarium_gates_nutrition_optimization/components/maternal_disorders.py

```diff
@@ -0,0 +1,53 @@
+import pandas as pd
+from vivarium.framework.engine import Builder
+from vivarium_public_health.disease import (
+    DiseaseModel,
+    RecoveredState,
+    SusceptibleState,
+)
+from vivarium_public_health.utilities import to_years
+
+from vivarium_gates_nutrition_optimization.components.disease import (
+    DiseaseState,
+    ParturitionSelectionState,
+)
+from vivarium_gates_nutrition_optimization.constants import data_keys, models
+from vivarium_gates_nutrition_optimization.constants.data_values import DURATIONS
+from vivarium_gates_nutrition_optimization.constants.metadata import (
+    ARTIFACT_INDEX_COLUMNS,
+)
+
+
+def MaternalDisorders():
+    cause = models.MATERNAL_DISORDERS_STATE_NAME
+    susceptible = ParturitionSelectionState(cause)
+    with_condition = DiseaseState(
+        cause,
+        get_data_functions={
+            "prevalence": lambda *_: 0.0,
+            "disability_weight": get_maternal_disorders_disability_weight,
+            "excess_mortality_rate": lambda *_: 0.0,
+            "dwell_time": lambda builder, cause: builder.time.step_size()(),
+        },
+    )
+    recovered = RecoveredState(cause)
+
+    susceptible.allow_self_transitions()
+    susceptible.add_transition(with_condition)
+    with_condition.allow_self_transitions()
```

Comment:
I think we could have just not allowed self transitions instead of setting the dwell time to the step-size? It's not worth changing anything, since this seems correct as well.